dnaStreaming/listener.py

```diff
@@ -23,13 +24,18 @@ def _initialize(self, config):
     def listen(self, on_message_callback, maximum_messages=DEFAULT_UNLIMITED_MESSAGES, subscription_id=None):
         limit_pull_calls = not (maximum_messages == self.DEFAULT_UNLIMITED_MESSAGES)
         pubsub_client = pubsub_service.get_client(self.config)
-
         subscription_id = subscription_id if subscription_id is not None else self.config.subscription()
 
         if subscription_id is None or subscription_id.strip() == '':
             raise Exception(
                 'No subscription specified. You must specify the subscription ID either through an environment variable, a config file or by passing the value to the method.')
 
+        streams_id = "-".join(subscription_id.split("-")[:4])
+        stream_url = self.config.streams_uri() + '/' + streams_id
+        r = requests.get(stream_url, headers=credentials_service.get_authentication_headers(self.config))
+        if r.json()['data']['attributes']['job_status'] == "EXCEEDED":
+            return "Your article limit has been exceeded. Please contact your customer service representitive for more info."
+
```

Comment:
I'd suggest moving this logic into a separate function that returns a boolean. Then you could do something like `if stream_limit_exceeded(subscription_id) return ....`. That way the logic can be reused by Cristian's new listen_async method

Reply:
Also the name of the status for exceeded streams is `DOC_COUNT_EXCEEDED`